core/src/main/java/com/oath/oak/ThreadIndexCalculator.java

```diff
@@ -0,0 +1,35 @@
+
+package com.oath.oak;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ThreadIndexCalculator {
+
+    private final AtomicInteger nextFreeIndex;
+    private final ConcurrentMap<Long, Integer> map;
+    static final int MAX_THREADS = 32;
+
+    private ThreadIndexCalculator() {
+        this.nextFreeIndex = new AtomicInteger(0);
+        map = new ConcurrentHashMap<>();
+    }
+
+
+    public int getIndex() {
+        long tid = Thread.currentThread().getId();
```

Comment:
because the jvm has no guarantee on thread ids.
So there are cases that there are 3 threads and the ids are 0,32,64 this causes all threads to go to same index.